azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java

```diff
@@ -482,13 +482,56 @@
     }
 
     /**
+     * The virtual machine definition stage with availability set.
+     *
+     * @param <T> The virtual machine definition in creatable stage.
+     */
+    interface DefinitionWithAvailabilitySet<T extends DefinitionCreatable> {
+        /**
+         * Specifies the name of the availability set to create, the virtual machine will be part of
+         * this availability set.
+         * <p/>
+         * Adding virtual machines running your application to an availability set ensures that during
+         * maintenance event at least one virtual machine will be available.
+         *
+         * @param name The name of the availability set
+         * @return The stage representing creatable VM definition
+         */
+        T withNewAvailabilitySet(String name);
+
+        /**
+         * Specifies an instance of AvailabilitySet.DefinitionCreatable representing the availability set
+         * to be created, the virtual machine will be part of this availability set.
+         * <p/>
+         * Adding virtual machines running your application to an availability set ensures that during
+         * maintenance event at least one virtual machine will be available.
+         *
+         * @param creatable The availability set in creatable stage
+         * @return The stage representing creatable VM definition
+         */
+        T withNewAvailabilitySet(AvailabilitySet.DefinitionCreatable creatable);
+
+        /**
+         * Specifies the name of an existing availability set under which this virtual machine needs to be
+         * added.
+         * <p/>
+         * Adding virtual machines running your application to an availability set ensures that during
+         * maintenance event at least one virtual machine will be available.
+         *
+         * @param name The name of an existing availability set
+         * @return The stage representing creatable VM definition
+         */
+        T withExistingAvailabilitySet(String name);
+    }
+
+    /**
      * The virtual machine definition stage with storage account.
      *
      * @param <T> The virtual machine definition in creatable stage.
```

Comment:
like before: param and return descriptions start with small letter and no period